pennylane/templates/subroutines/qmc.py

```diff
@@ -326,32 +325,47 @@ def circuit():
         >>> (1 - np.cos(np.pi * phase_estimated)) / 2
         0.4327096457464369
     """
-    if isinstance(probs, np.ndarray) and probs.ndim != 1:
-        raise ValueError("The probability distribution must be specified as a flat array")
-
-    dim_p = len(probs)
-    num_target_wires = float(np.log2(2 * dim_p))
-
-    if not num_target_wires.is_integer():
-        raise ValueError("The probability distribution must have a length that is a power of two")
-
-    num_target_wires = int(num_target_wires)
-
-    target_wires = Wires(target_wires)
-    estimation_wires = Wires(estimation_wires)
-
-    if num_target_wires != len(target_wires):
-        raise ValueError(
-            f"The probability distribution of dimension {dim_p} requires"
-            f" {num_target_wires} target wires"
-        )
-
-    A = probs_to_unitary(probs)
-    R = func_to_unitary(func, dim_p)
-    Q = make_Q(A, R)
-
-    qml.QubitUnitary(A, wires=target_wires[:-1])
-    qml.QubitUnitary(R, wires=target_wires)
-    qml.templates.QuantumPhaseEstimation(
-        Q, target_wires=target_wires, estimation_wires=estimation_wires
-    )
+    num_params = 3
```

Comment:
The design that `Operations` have arguments that are parameters is a historical fact, and when rewriting templates as operations it sometimes feels a bit arbitrary to determine which is which. But so far I thought we should only treat inputs such as weights and features (in embeddings) as parameters. Not sure what the true parameters would be here? I am wondering why you decided to use `probs`, `func` and `target_wires` here, but not `estimation_wires`?